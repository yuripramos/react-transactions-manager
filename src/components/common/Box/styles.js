import styled, { css } from "styled-components";
import { white, grey10 } from "../../../styles/settings";
import { rem, media } from "../../../styles/tools/index";
import { isResponsive } from "../../../utils/getResolution";

export const Title = styled.span`
  color: ${grey10};
  font-size: ${rem(11)};
  font-weight: bold;
  line-height: 1.45;
  text-transform: uppercase;
`;

export const IconContainer = styled.span`
  align-items: center;
  background: ${white};
  border-radius: ${rem(4)};
  color: ${grey10};
  display: inline-flex;
  height: ${rem(24)};
  justify-content: center;
  margin-right: ${rem(9)};
  width: ${rem(24)};
`;

export const Header = styled.div`
  align-items: center;
  display: flex;
`;

export const Wrapper = styled.div`
  display:inline-flex;
  flex-wrap: wrap;
  ${({ noTitle }) =>
  noTitle &&
    css`
      ${media.xs(css`
        transform: translateY(0px);
      `)};
      ${media.md(css`
        transform: translateY(10px);
      `)};
  `};
  ${({ width }) =>
  width &&
  css`
    width: ${isResponsive() ? rem(width/0.7) : rem(width)};
  `};
`;

