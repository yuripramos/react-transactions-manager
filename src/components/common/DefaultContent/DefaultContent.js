import React from "react";
import { func, node, string, array } from "prop-types";
import { generate } from "shortid";

import {
  Wrapper,
  IconWrapper,
  PrimaryText,
  SecondaryText,
} from "./styles";

function DefaultContent({
  Icon,
  primaryText,
  secondaryTexts,
  children,
}) {
  return (
    <Wrapper>
      {Icon && (
        <IconWrapper>
           <Icon />
        </IconWrapper>
      )}
      <PrimaryText>{primaryText}</PrimaryText>
      {secondaryTexts &&
        secondaryTexts.map(item => (
          <SecondaryText key={generate()}>{item}</SecondaryText>
        ))}
      {children}
    </Wrapper>
  );
}

DefaultContent.defaultTypes = {
  secondaryTexts: [],
  Icon: () => {},
  children: null
};

DefaultContent.propTypes = {
  Icon: func,
  primaryText: string.isRequired,
  secondaryTexts: array,
  children: node
};

export default DefaultContent;
