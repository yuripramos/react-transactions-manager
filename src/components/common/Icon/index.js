import React from "react";
import { string } from "prop-types";

/*ICONS*/
import Flag from "./Flag";
import Star from "./Star";
import NotFound from "./NotFound";
import Truck from "./Truck";
import To from "./To";
import Airport from "./Airport";
import Cfs from "./Cfs";
import LiftGate from "./LiftGate";
import PalletJack from "./PalletJack";
import Lumper from "./Lumper";
import Menu from "./Menu";
import MenuClose from "./MenuClose";

const IconGenerator = props => {
  switch (props.name) {
    case "Flag":
      return <Flag {...props} />;
    case "Star":
      return <Star {...props} />;
    case "NotFound":
      return <NotFound {...props} />;
    case "Truck":
      return <Truck {...props} />;
    case "To":
      return <To {...props} />;
    case "airport":
      return <Airport {...props} />;
    case "cfs":
      return <Cfs {...props} />;
    case "lift_gate":
      return <LiftGate {...props} />;
    case "pallet_jack":
      return <PalletJack {...props} />;
    case "lumper":
      return <Lumper {...props} />;
    case "Menu":
      return <Menu {...props} />;
    case "MenuClose":
      return <MenuClose {...props} />;
    default:
      return;
  }
};

IconGenerator.propTypes = {
  name: string
};

export default IconGenerator;
