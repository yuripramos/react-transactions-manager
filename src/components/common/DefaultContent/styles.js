import styled, { css } from "styled-components";
import { rem, media } from "../../../styles/tools";
import { black30, black50 } from "../../../styles/settings";

export const Wrapper = styled.div`
  text-align: center;
  padding: ${rem(25)} ${rem(10)};

  ${media.md(css`
    padding: ${rem(50)} ${rem(10)};
  `)};
`;

export const IconWrapper = styled.div`
  margin-bottom: ${rem(11)};

  ${media.md(css`
    margin-bottom: ${rem(15)};
  `)};
`;

export const PrimaryText = styled.h3`
  font-size: ${rem(16)};
  letter-spacing: ${rem(0.5)};
  text-align: left;
  color: ${black30};
  margin-bottom: ${rem(10)};
  display: inline-block;
  text-align: center;

  ${media.md(css`
    font-size: ${rem(28)};
    margin-bottom: ${rem(20)};
  `)};
`;
export const SecondaryText = styled.p`
  font-size: ${rem(13)};
  letter-spacing: ${rem(0.6)};
  color: ${black50};
  line-height: 1.45;
  text-align: center;

  ${media.md(css`
    font-size: ${rem(22)};
    line-height: 1.45;
  `)};
`;
