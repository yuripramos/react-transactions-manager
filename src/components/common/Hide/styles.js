import styled, { css } from "styled-components";
import { media } from "../../../styles/tools";

export const HideStyle = styled.div`
  display: none;

  ${({ below }) =>
    below &&
    media[below](css`
      display: block;
    `)};

  ${({ above }) =>
    above &&
    css`
      display: block;

      ${media[above](css`
        display: none;
      `)};
    `};
`;
