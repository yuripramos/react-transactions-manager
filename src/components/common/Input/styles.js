import styled, { css } from "styled-components";
import { media, rem } from "../../../styles/tools";

import {
  white,
  grey30,
  black50,
  darkBlue,
  lightestBlue,
  red,
  grey180
} from "../../../styles/settings";

export const IconContainer = styled.div`
  position: absolute;
  top: 50%;
  left: 0;
  width: ${rem(55)};
  display: flex;
  justify-content: center;
  align-items: center;
  transform: translateY(-50%);
  color: ${black50};
  transition: 0.3s color;
  pointer-events: none;
`;

export const Prefix = styled.span`
  position: absolute;
  padding: ${rem(15)} ${rem(0)} ${rem(0)} ${rem(10)};
  pointer-events: none;
  font-family: Lato;
  font-size: ${rem(13)};
  color: ${grey180};
`;

export const Tooltip = styled.div`
  visibility: visible;

  ${({ hide }) =>
    hide &&
    css`
      visibility: hidden;
    `};
`;

export const Label = styled.label`
  position: absolute;
  left: ${rem(52)};
  top: 50%;
  transform: translateY(-50%);
  pointer-events: none;
  transition: 0.3s top, 0.2s color, 0.3s font-size;
  font-family: Lato;
  font-size: ${rem(13)};
  letter-spacing: ${rem(0.1)};
  color: ${black50};
  ${({ tinyLabels }) =>
    tinyLabels &&
    css`
      font-size: ${rem(10)};
      left: ${rem(10)};
      top: 50%;
      transform: translateY(-50%);
    `};
`;

const revealContent = css`
  ~ ${Label} {
    top: ${rem(22)};
    font-size: ${rem(11)};
    letter-spacing: ${rem(0.1)};
    text-align: left;
    color: ${darkBlue};
  }
`;

const revealContent2 = css`
  ~ ${Label} {
    top: ${rem(10)};
    font-family: Lato Bold;
    font-size: ${rem(9)};
    letter-spacing: ${rem(0.1)};
    text-align: left;
    color: ${darkBlue};
  }
`;

const revealContent3 = css`

  ~ ${Label} {
    top: ${rem(12)};
    color: ${black50};
  }
`;

export const InputField = styled.input`
  width: 100%;
  height: ${rem(46)};
  border-radius: ${rem(4)};
  background-color: ${white};
  border: solid ${rem(1)} ${grey30};
  font-size: ${rem(13)};
  letter-spacing: ${rem(0.1)};
  color: ${black50};
  padding: 0 ${rem(52)};
  transition: 0.3s padding, 0.3s border;
  box-shadow: none;

  ::-ms-clear {
    display: none !important;
  }
  ${({ prefix }) =>
    prefix &&
    css`
      padding-left: ${rem(32)} !important;
    `};

  :focus {
    outline: none;
    border: solid ${rem(1)} ${darkBlue};
  }

  ${({ tinyLabels }) =>
    tinyLabels &&
    css`
      padding: 0 ${rem(9.2)};
    `};

  :focus {
    ${({ tinyLabels }) => (tinyLabels ? revealContent2 : revealContent)};
  }
  ${({ isEmpty }) => !isEmpty && revealContent};
  ${({ isEmpty, tinyLabels }) => !isEmpty && tinyLabels && revealContent2};
  :not(:focus) {
    ${({ tinyLabels, isEmpty }) => tinyLabels && !isEmpty && revealContent3};
  }
`;

export const InputWrapper = styled.div`
  width: 100%;
  display: inline-block;
  position: relative;
  height: ${rem(38)};
  margin-right: ${rem(25)};
  input:not(:focus) {
    ${({ valid }) =>
      valid === false &&
      css`
        border-color: ${red};
      `};
  }
  ${({ width }) =>
    width &&
    css`
      width: ${width}%;
    `};

  ~ input {
    background-color: #ccc;
  }
`;

export const WrapperDayPicker = styled.div`
  margin-right: ${rem(10)};
  .DayPickerInput > input {
    width: ${rem(60)};
    height: ${rem(46)};
    border-radius: ${rem(4)};
    background-color: ${white};
    border: solid ${rem(1)} ${grey30};
    font-size: ${rem(11)};
    letter-spacing: ${rem(0.1)};
    color: ${black50};
    padding: 0 ${rem(20)};
    transition: 0.3s padding, 0.3s border;
    box-shadow: none;

    ::-ms-clear {
      display: none !important;
    }

    :focus {
      outline: none;
      border: solid ${rem(1)} ${darkBlue};
    }
  }

  .DayPickerInput-OverlayWrapper {
    position: absolute;
    background: ${lightestBlue};
    left: 78px;
    top: -15px;
    ${media.md(css`
      left: 166px;
       top: -30px;
    `)};
  }

  .DayPickerInput-Overlay {
    position: absolute;
    left: 0;
    z-index: 999;

    background: ${lightestBlue};
    box-shadow: 0 2px 5px rgba(0, 0, 0, 0.15);
  }
`;
