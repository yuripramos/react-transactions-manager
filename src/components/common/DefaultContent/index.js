import DefaultContent from "./DefaultContent";

export default DefaultContent;
