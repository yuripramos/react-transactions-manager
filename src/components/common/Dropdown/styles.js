import styled, { css } from "styled-components";
import { media, rem } from "../../../styles/tools";
import {
  white,
  grey30,
  black50,
  darkBlue,
  green,
  red
} from "../../../styles/settings";
import { Label } from "../../common/Input/styles";

export const DropdownField = styled.select`
  width: 100%;
  height: ${rem(48)};
  border-radius: ${rem(4)};
  background-color: ${white};
  border: solid ${rem(1)} ${grey30};
  font-family: Lato;
  font-size: ${rem(13)};
  letter-spacing: ${rem(0.8)};
  color: ${black50};
  padding: ${rem(15)} ${rem(55)} 0 ${rem(6.4)};
  transition: 0.3s padding, 0.3s border;
  box-shadow: none;

  ::-ms-clear {
    display: none !important;
  }

  :focus {
    outline: none;
    border: solid ${rem(1)} ${darkBlue};
  }

  ${({ isEmpty, tinyLabels }) => !isEmpty && tinyLabels && revealContent2};
  :not(:focus) {
    ${({ tinyLabels, isEmpty }) => tinyLabels && !isEmpty && revealContent3};
  }

  ${({ hasLabel }) =>
    !hasLabel &&
    css`
      height: auto;
      padding: ${rem(8)} 0 ${rem(8)} ${rem(15)};
      border-radius: ${rem(20)};
      appearance: none;
    `};
`;

export const DropdownWrapper = styled.div`
  letter-spacing: ${rem(0.1)};
  display: inline-block;
  position: relative;
  margin: 0 ${rem(6)} ${rem(6)} 0;

  select:not(:focus) {
    ${({ valid }) =>
      valid === false &&
      css`
        border-color: ${red};
      `};
  }

  ${({ width }) =>
    width &&
    css`
      width: ${width}%;
    `};
`;

const revealContent2 = css`
  ~ ${Label} {
    top: ${rem(10)};
    font-family: Lato Bold;
    font-size: ${rem(11)};
    letter-spacing: ${rem(0.1)};
    text-align: left;
    color: ${darkBlue};
  }
`;

const revealContent3 = css`
  ~ ${Label} {
    top: ${rem(12)};
    color: ${black50};
  }
`;

export const IconContainer = styled.div`
  position: absolute;
  top: 50%;
  left: 0;
  width: ${rem(55)};
  display: flex;
  justify-content: center;
  align-items: center;
  transform: translateY(-50%);
  color: ${black50};
  transition: 0.3s color;
  pointer-events: none;
`;
