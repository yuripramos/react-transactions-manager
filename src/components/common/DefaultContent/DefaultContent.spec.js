import React from "react";
import { shallow } from "enzyme";

import DefaultContent from "./DefaultContent";
import { black30 } from "../../../styles/settings";
import Icon from "../Icon";

const props = {
  Icon: () => <Icon type="Attention" color={black30} />, // eslint-disable-line
  primaryText: "primary text",
  secondaryTexts: ["text", "more text"]
};

describe("DefaultContent component", () => {
  it("should match snapshot", () => {
    expect(shallow(<DefaultContent {...props} />)).toMatchSnapshot();
  });
});
