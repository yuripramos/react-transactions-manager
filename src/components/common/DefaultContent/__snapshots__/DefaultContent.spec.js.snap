// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultContent component should match snapshot 1`] = `
<styled.div>
  <styled.div>
    <Icon />
  </styled.div>
  <styled.h3>
    primary text
  </styled.h3>
  <styled.p
    key="hash"
  >
    text
  </styled.p>
  <styled.p
    key="hash"
  >
    more text
  </styled.p>
</styled.div>
`;
